File: src/main/java/com/rust/model/PipelineConfig.java
package com.rust.model;

import glm.vec._4.Vec4;

public class PipelineConfig {
    public double oc;
    public double cc;
    public double sc;
    public double rh;
    public double test;
    public double temperature;
    public double growTime = 40;
    public int textureWidth = 400;
    public int textureHeight = 400;
    public boolean generateImage = false;
    public String diffPath = "diff.png";
    public Vec4[][] backTexture;
    public float[][] metallic;
    public float[][] roughness;
    public boolean useCurvature = true;
    public float[][] curvature = null;
}

File: Source/Core/PipelineConfig.cpp
#include "PipelineConfig.h"

PipelineConfig::PipelineConfig()
{
    oc = 0;
    cc = 0;
    sc = 0;
    rh = 0;
    test = 0;
    temperature = 0;
    growTime = 40;
    textureWidth = 800;
    textureHeight = 800;
    generateImage = false;
    diffPath = "diff.png";

    backTexture = vector<vector<vec4>>(textureHeight, vector<vec4>(textureWidth, vec4()));
    metallic = vector<vector<float>>(textureHeight, vector<float>(textureWidth, 0));
    roughness;
    useCurvature = true;
    curvature;
}
